srml/contracts/src/rent.rs

```diff
@@ -143,17 +148,26 @@ fn try_evict_or_and_pay_rent<T: Trait>(
 		// The contract cannot afford the rent payment and has a balance above the subsistence
 		// threshold, so it leaves a tombstone.
 
-		// Note: this operation is heavy.
-		let child_storage_root = runtime_io::child_storage_root(&contract.trie_id);
-
+		let p_key = prefixed_trie_id(&contract.trie_id);
+		let (o_ct, child_storage_root) = if let Some(child_trie) = child::child_trie(p_key.as_ref()) {
+			// Note: this operation is heavy.
+			let child_storage_root = runtime_io::child_storage_root(&child_trie);
+			(Some(child_trie), child_storage_root)
+		} else {
+			// Note that this will fail as soon as we support multiple type
+			// of hashing for child trie.
+			let child_storage_root =
+				<<T as system::Trait>::Hashing as HashT>::enumerated_trie_root(&[]);
+			let child_storage_root_ref: &[u8] = child_storage_root.as_ref();
+			(None, child_storage_root_ref.to_vec())
```

Comment:
Hmm, in what case would the child trie not exist? Since it's a live contract I think it's OK to return an error here if the child trie doesn't exist.

Reply:
It can happen if a contract id got all its content removed (the child trie is deleted if its associated content length is 0 (I added this condition somewhere in one of my latest commit for obvious reason)).